codec/acceptance_testing.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2021 - The Event Horizon authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codec
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+
+	eh "github.com/looplab/eventhorizon"
+	"github.com/looplab/eventhorizon/mocks"
+)
+
+// EventCodecAcceptanceTest is the acceptance test that all implementations of
+// Codec should pass. It should manually be called from a test case in each
+// implementation:
+//
+//   func TestEventCodec(t *testing.T) {
+//       c := EventCodec{}
+//       expectedBytes = []byte("")
+//       eventbus.AcceptanceTest(t, c, expectedBytes)
+//   }
+//
+func EventCodecAcceptanceTest(t *testing.T, c eh.EventCodec, expectedBytes []byte) {
```

Comment:
Maybe it would be good to add some more tests with for example nested structs, nil definitions, single bools etc.

Reply:
Sounds good!